numpy/lib/arraysetops.py

```diff
@@ -334,7 +337,8 @@ def _unique1d(ar, return_index=False, return_inverse=False,
         aux = ar
     mask = np.empty(aux.shape, dtype=np.bool_)
     mask[:1] = True
-    if aux.shape[0] > 0 and aux.dtype.kind in "cfmM" and np.isnan(aux[-1]):
+    if (equal_nans and aux.shape[0] > 0 and aux.dtype.kind in "cfmM" and 
+    np.isnan(aux[-1])):
```

Comment:
This always looks weird: either it lines up with the line below, or is over-indented. I forget what black does here...
```suggestion
            np.isnan(aux[-1])):
```